test/cxx_api_tests/RoadRunnerMapTests.h

```diff
@@ -0,0 +1,258 @@
+//
+// Created by Ciaran on 09/12/2021.
+//
+
+#include "gtest/gtest.h"
+#include "rrRoadRunnerMap.h"
+#include "rrConfig.h"
+#include "TestModelFactory.h"
+#include <filesystem>
+
+using namespace rr;
+
+class RoadRunnerMapTests : public ::testing::Test {
+public:
+    std::vector<std::string> sbml;
+
+    RoadRunnerMapTests() {
+#ifndef SBMLTestSuiteRoot
+        throw std::logic_error("The SBMLTestSuiteRoot variable is not defined even though it has been "
+                               "set with target_compile_definitions")';
+#endif
+        if (!std::filesystem::exists(SBMLTestSuiteRoot)) {
+            throw std::invalid_argument("Path to sbml test suite does not exist");
+        }
+        Logger::setLevel(Logger::LOG_WARNING);
+
+        // little easy list of sbmls
+        std::vector<std::string> input({
+                                               OpenLinearFlux().str(),
+                                               SimpleFlux().str(),
+                                               Brown2004().str(),
+                                               Venkatraman2010().str()
+                                       });
+        sbml.swap(input);
+    }
+
+    std::string constructSTSXmlFile(int caseId, int level, int version) {
+        std::ostringstream caseOs;
+        std::ostringstream fname;
+
+        caseOs << std::setfill('0') << std::setw(5) << caseId;
+        fname << caseOs.str() << "-sbml-l" << level << "v" << version << ".xml";
+
+        std::filesystem::path root = SBMLTestSuiteRoot;
+        std::filesystem::path caseDir = root / caseOs.str();
+        std::filesystem::path xmlFile = caseDir / fname.str();
+        if (!std::filesystem::exists(xmlFile)) {
+            std::ostringstream os;
+            os << "file not found : " << xmlFile;
+            throw std::logic_error(os.str());
+        }
+        return xmlFile.string();
+    }
+
+    std::vector<std::string> getFirstNModelsFromSTS(int N, int start = 1) {
+        if (start < 1){
```

Comment:
Why define this both here in the .h file and then again in the .cpp file?

Reply:
And indeed, the .h file and the .cpp file are identical.

Reply:
So, I made my last commit quite hastily because I know you would be reviewing the content and I wanted you to have the latest bug fix. Therefore, these got committed by mistake, in transit to `<roadrunner_root>/test/sbml-test-suite/STS`.